### PR TITLE
Task dashboard

### PR DESCRIPTION
**Pull request by Programmingrelated1999**
after task deletion
This is a PR for Task Deletion Dashboard. The new features are: 

* There will be additional section in Task: Status
* The Status is for admin and creator positions and will have delete the task button
* If the delete task button is clicked, the task will be deleted from database.

I have added new Status section and delete button in Team Member Task component. Moreover, I have added the delete request from frontend in the reducer file in Team Member Task folder. 

Below are screenshots of the delete button.
before task deletion
task delete button clicked

### CODE REVIEW

**Comment by EvianTan** [comment]
Hi @Programmingrelated1999 ,

Awesome job! 
Finally figured out this PR is only for "create ability for admin/owner users to delete tasks from dashboard", so I updated our actions list as below to be more accurate. I split your action 2.3.8 from 2.3.5 as two tasks.
image

✅ Verified changes:
1. Added a new column on dashboard task tab called "Status"
2. Only admin/owner can see the "Delete" buttons for tasks
3. Tasks can be totally removed from dashboard/WBS page and database

💡 Things can be improved:
1. The little red cross remove sign
For the consistency of signs in our app, it's always better to pick one from [this library](https://fontawesome.com/v4/icons/). You can also find that from our codes, most of the signs are start with "fa ...", that's because they both from this same library. Try to use this [remove](https://fontawesome.com/v4/icon/times) sign instead. Also here is a [link](https://www.freecodecamp.org/news/how-to-add-font-awesome-icons-to-your-buttons/) for use font awesome icon in button
2. Alignment of the task table header and data
image

❓ Question:
Just curious about this PR, why it shows so many unrelated commits?? I hit this things once, but didn't know how did I trigger that 🥲

**Comment by EvianTan** [comment]
Oh btw, for naming working branch, it's better to use something like "Yourfirstname-the_feature_you_are_working". Like this branch could be "Kaung-create_task_deletion_on_dashboard". It will be quick for others and yourself to find the branch from tons of branches, and the more meaningful branch name can give others more info on the first glance.

**Comment by one-community** [comment]
Following up on this, does it still need to be reviewed and integrated?

**Comment by EvianTan** [comment]
> Following up on this, does it still need to be reviewed and integrated?

Yes it still need to reviewed. I'm reviewing it now.

**Comment by Red4lpha** [comment]
I cannot accept this PR, I don't think this branch was properly updated with development branch when it was being worked on. My changes to WBS task action menu got removed and it now that function is completely broken:
Week11-PR-brokenFeature

This leads me to believe other developers recent changes have been possibly broken as well
Furthermore, the red X for is not align well with it's column and does not seemingly work. Where's the buttons to delete it? and the popup message only works half the time.
Week11-PRBrokenFeature2

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx [hunk 1]
```diff
@@ -413,27 +413,11 @@ const Task = props => {
               <i className="fa fa-book" aria-hidden="true"></i>
             </td>
           </tr>
-
```
Comment (Red4lpha) [inline src/components/Projects/WBS/WBSDetail/Task/Task.jsx:416]: Why was my feature deletion? Was this properly updated with the development branch before PR?

**Comment by pedroelton** [comment]
> I cannot accept this PR, I don't think this branch was properly updated with development branch when it was being worked on. My changes to WBS task action menu got removed and it now that function is completely broken: Week11-PR-brokenFeature
> 
> This leads me to believe other developers recent changes have been possibly broken as well Furthermore, the red X for is not align well with it's column and does not seemingly work. Where's the buttons to delete it? and the popup message only works half the time. Week11-PRBrokenFeature2



> I cannot accept this PR, I don't think this branch was properly updated with development branch when it was being worked on. My changes to WBS task action menu got removed and it now that function is completely broken: Week11-PR-brokenFeature
> 
> This leads me to believe other developers recent changes have been possibly broken as well Furthermore, the red X for is not align well with it's column and does not seemingly work. Where's the buttons to delete it? and the popup message only works half the time. Week11-PRBrokenFeature2

I had a look here, and someone doubled this line by mistake.
Around line 567 it was supposed to be like this:
```
          </tr>
        </>
      ) : null}
    </>
```

And it was like this:
```
          </tr>
          ) : null}
        </>
      ) : null}
    </>
```

**Review by EvianTan — CHANGES_REQUESTED** [review]

**Comment by EvianTan** [comment]
Hey @Programmingrelated1999 ,

1. Please don't include the formatting/ESLint/Prettier changes in this PR since it is a functionality change one, and it would be really hard for people to review the actual code changes you made. 
2. Like Kevin mentioned, his feature works well on Main/Dev but totally broken on this branch. Please fetch the latest Dev code and make sure you don't break any existing functionalities. We don't want to introduce bug again in such a hard time.

**Comment by Filipe-Oliv** [comment]
Is this PR still being reviewed?

**Comment by one-community** [comment]
This one needs to be taken over and wrapped up by someone else.

**Comment by Filipe-Oliv** [comment]
Working on it

**Comment by Filipe-Oliv** [comment]
The WBS task action menu that was removed and the function was stopped is working again. Could you chech it, please?

Before:


After:

[02-18-2023 (2).webm](https://user-images.githubusercontent.com/114119668/219869816-7fd58584-e5b3-45de-acc1-80681fe4ffb1.webm)

That's all for now. On Monday I'll keep working on it.

**Comment by PrabhjotSembhi** [comment]
@Filipe-Oliv Let me know once it's ready to review. Thanks

**Pull request closed without merging**